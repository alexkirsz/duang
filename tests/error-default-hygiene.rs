#![allow(non_snake_case)]
#![allow(unused_variables)]
mod bar {
  use duang::duang;

  pub static NUM: i32 = 42;

  duang!(
  pub fn foo(a: i32 = NUM) -> i32
  {
    a
  }
  );
}

fn main() {
  use bar::foo;
  foo!();
}
