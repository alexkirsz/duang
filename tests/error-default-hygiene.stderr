error[E0425]: cannot find value `NUM` in this scope
  --> $DIR/error-default-hygiene.rs:9:23
   |
9  |   pub fn foo(a: i32 = NUM) -> i32
   |                       ^^^ not found in this scope
...
18 |   foo!();
   |   ------- in this macro invocation
help: possible candidate is found in another module, you can import it into scope
   |
3  | use crate::bar::NUM;
   |

For more information about this error, try `rustc --explain E0425`.
